src/components/tooltip/_tooltip.scss

```diff
@@ -1,31 +1,129 @@
-@import '~styles/variables';
-
-.tooltip {
+.player .tooltip {
+  position: relative;
   display: inline-block;
-  height: 22px;
-  border-radius: 4px;
-  background-color: $tooltip-bg-color;
-  padding: 3px 13px;
-  color: $tooltip-color;
-  font-size: 13px;
-  font-weight: bold;
-  line-height: 16px;
-  box-shadow: 0 0 8px 0 rgba(0, 0, 0, 0.3);
-  position: absolute;
-  transform: translate3d(-50%, 0, 0);
-  left: 50%;
-  bottom: 15px;
-
-  &:after {
-    content: ' ';
-    display: block;
+  .tooltip-label {
+    width: max-content;
+    visibility: hidden;
+    background-color: $tooltip-bg-color;
+    color: $tooltip-color;
+    text-align: center;
+    padding: 4px 6px;
+    border-radius: 4px;
     position: absolute;
-    width: 10px;
-    height: 10px;
-    background-color: #fff;
-    border-radius: 3px;
-    transform: rotate(45deg) translate3d(0%, 50%, 0);
-    left: 50%;
-    bottom: -1px;
+    line-height: normal;
+    font-weight: bold;
+    font-size: 13px;
+    z-index: 2;
+    opacity: 0;
+    transition: opacity 1s;
+    &:after {
+      content: '';
+      position: absolute;
+      border-width: 5px;
+      border-style: solid;
+      border-radius: 3px;
+      color: $tooltip-bg-color;
+      z-index: -1;
+    }
+    &.tooltip-top {
+      bottom: $tooltip-margin;
+      left: 50%;
+      transform: translate(-50%);
+      &:after {
+        bottom: -$tooltip-arrow-height;
+        left: 50%;
+        transform: translate(-50%) rotate(45deg) skew(5deg, 5deg);
+      }
+    }
+    &.tooltip-top-left {
```

Comment:
in generally `.tooltip-top-left` should be a combination of `.tooltip-top` and `.tooltip-left` as we do in the watermark component. does it applicable here?
any way, `.tooltip-top-left` and `.tooltip-top-right` should use `.tooltip-top` rules. only overide the different rule. 
try to reuse the rules as much as possible

Reply:
not possible as we discussed